text/0000-pack-direct-podman-call.md

```diff
@@ -0,0 +1,86 @@
+# Meta
+[meta]: #meta
+- Name: Pack direct podman call
+- Start Date: 2023-06-16
+- Author(s): dvaumoron
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+A flag `--container-engine` will be added to the `pack` CLI, the allowed flag value will be `docker` (the current functioning) and `podman` to call podman directly (without socket call)
+
+# Definitions
+[definitions]: #definitions
+
+N/A
+
+# Motivation
+[motivation]: #motivation
+
+This will allow to use pack as a standalone on the OS supported by podman (currently Linux, as Windows and macOS need podman machine), making possible to use pack without installing docker or podman (a tool like skopeo will still be necessary to send image in registries)
```

Comment:
Can we expand on the motivation to explain what the difference is between the proposed mode of operation and the existing mode of operation?

```suggestion
`pack` currently requires an orchestration client API available via a socket.  Both `docker` and `podman` are compatible with the [socket API](https://docs.docker.com/engine/api/v1.24/).  This RFC proposes that `pack` is extended to support a "daemonless" mode.  In a "daemonless" mode `pack` will directly use an orchestration client API.  This avoids running a `docker` or `podman` daemon and avoids opening up a local socket.  This will allow to use pack as a full-userspace, standalone application on the OS supported by podman (currently Linux, as Windows and macOS need podman machine).
```